### ISSUE TITLE
Custom notification messages

### ISSUE THREAD

**Issue by MMauro94** [offen/docker-volume-backup#59]
* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request

Right now there is no way of customizing error and success messages for notifications. As I have multiple backup containers, I struggle to identify at a glance which backup has run, since I have to read the log. Also I don't need the full log on success notifications.

It would be nice having the ability to customize these messages, while injecting useful information. Something like this:

```
SUCCESS_NOTIFICATION_MESSAGE={CONTAINER_LABEL} backup succeeded at {BACKUP_TIME}
```

Thoughts? If you're interested in implementing this feature I can produce a list of useful variables to be injected.

Open questions:
1. Should the same feature be available for the notification title?
2. Should it be possible to provide different messages for different shoutrrr URLs? (For example send the full log via email but only a short message in Telegarm) For this one I'd argue no, since it would become too complex to configure by environment variables, but I just thought to put it out here.

**Comment by m90** [comment]
Thanks for submitting this idea. While I probably would not use it myself I can see how it can be useful, so I would be happy if anyone wants to implement it and am happy to help with reviewing/merging PRs.

---

A few thoughts on the implementation from my end: Golang has a `text/template` package in the standard libary which would be a good fit for this I guess. Right now a notification is just hardcoded here: https://github.com/offen/docker-volume-backup/blob/3bb99a711754ba53f549c70c8d0914b6007e329d/cmd/backup/main.go#L286-L296

This could however be turned into a standalone template file something like this (I have not used the package in ages, so this might be slightly wrong syntactically, but you will get the idea):

```
{{ define "title_failure" }}
Failure running docker-volume-backup at {{ .StartTime }}
{{ end }}

{{ define "body_failure" }}
Running docker-volume-backup failed with error: {{ .Error }}
Log output of the failed run was:

{{ .LogOutput }}
{{ end }}

// this would also define `title_success` and `body_success`
```

This file could be read at runtime from e.g. `/etc/volume-backup.d/00notifications.tmpl`. Users who wanted to customize these messages could now mount a `/etc/volume-backup.d/10override.tmpl` (either at runtime or by inheriting from the Docker image) file that can selectively overwrite any of the templates defined in the default templates.

The init procedure of the command would then read and parse all files in `/etc/volume-backup.d` and use the last occurrence of each template definition.

Do you think this makes sense?


About your questions:
1. yes, definitely
2. i agree this would become to complicated to configure

**Comment by m90** [comment]
This is implemented as of `v2.11.0`
